flow-server/src/main/java/com/vaadin/flow/dom/impl/StyleAttributeHandler.java

```diff
@@ -75,30 +73,62 @@ public void setAttribute(Element element, String attributeValue) {
      */
     public static LinkedHashMap<String, String> parseStyles(
             String styleString) {
-        CollectingCSSParseErrorHandler errorCollector = new CollectingCSSParseErrorHandler();
-        CSSDeclarationList parsed = CSSReaderDeclarationList.readFromString(
-                styleString, ECSSVersion.LATEST, errorCollector);
-        if (errorCollector.hasParseErrors()) {
-            throw new IllegalArgumentException(String
-                    .format(ERROR_PARSING_STYLE, styleString, errorCollector
-                            .getAllParseErrors().get(0).getErrorMessage()));
-        }
-        if (parsed == null) {
-            // Did not find any styles
-            throw new IllegalArgumentException(String.format(
-                    ERROR_PARSING_STYLE, styleString, "No styles found"));
+        try {
+            LinkedHashMap<String, String> parsedStyles = new LinkedHashMap<>();
+            StringBuilder nameBuffer = new StringBuilder();
+            StringBuilder valueBuffer = new StringBuilder();
+            boolean nameRead = false;
+            int parenthesisOpen = 0;
+            for (int i = 0; i < styleString.length(); i++) {
+                char c = styleString.charAt(i);
+                if (nameRead) {
+                    boolean valueTerminated = false;
+                    if (c == PARENTHESIS_OPEN) {
+                        parenthesisOpen++;
+                    } else if (c == PARENTHESIS_CLOSED) {
+                        parenthesisOpen--;
+                    } else if (parenthesisOpen == 0 && c == SEMICOLON) {
+                        valueTerminated = true;
+                    }
+                    if (valueTerminated) {
+                        addRule(nameBuffer, valueBuffer, parsedStyles);
+                        nameBuffer = new StringBuilder();
+                        valueBuffer = new StringBuilder();
+                        nameRead = false;
+                    } else {
+                        valueBuffer.append(c);
+                    }
+                } else {
+                    if (c == ":".charAt(0)) {
```

Comment:
```suggestion
                    if (c == ':') {
```
This could also have the `COLON` constant as the other parts.